server/lib/chat/chat-service.ts

```diff
@@ -490,43 +533,44 @@ export default class ChatService {
     })
   }
 
-  async getChannelInfo(channelId: SbChannelId, userId: SbUserId): Promise<ChannelInfo> {
-    const channelInfo = await getChannelInfo(channelId)
+  async getChannelInfo(channelId: SbChannelId, userId: SbUserId): Promise<GetChannelInfoResponse> {
+    const [channelInfo, userChannelEntry] = await Promise.all([
+      getChannelInfo(channelId),
+      getUserChannelEntryForUser(userId, channelId),
+    ])
+    const isUserInChannel = Boolean(userChannelEntry)
 
     if (!channelInfo) {
       throw new ChatServiceError(ChatServiceErrorCode.ChannelNotFound, 'Channel not found')
     }
 
-    let userCount
-    if (!channelInfo.private || this.state.users.get(userId)?.has(channelId)) {
-      userCount = channelInfo.userCount
-    }
-
+    const channelSummary = toChannelSummary(channelInfo)
     return {
-      id: channelInfo.id,
-      name: channelInfo.name,
-      private: channelInfo.private,
-      official: channelInfo.official,
-      userCount,
+      channelInfo: channelSummary.channelInfo,
```

Comment:
Kinda feels like if we have the `joinedChannelInfo` here and they've joined, we may as well send that here too? Not sure why we just discard it

Reply:
Yeah, I was thinking about that too. But then I felt like that would just be extra information for this API. Maybe if we renamed this API to `getChannelDetails` it would make more sense?

Kinda similarly how we don't send `detailedChannelInfo` and `joinedChannelInfo` in channel mentions, even though we have that information there as well.

Reply:
I see no significant difference between `getChannelInfo` and `getChannelDetails` as a name, these are the same thing. I just don't see why we're grabbing data from the DB that can be shown to the client, then throwing it away and making them request this data from somewhere else later (which will then have to re-make the exact same query to get it)

This is not the same as not sending `detailedChannelInfo` and `joinedChannelInfo` in channel mentions, those are being sent to people who we do not know have permission to see those things (and checking that for every message is prohibitively expensive).